_inc/lib/widgets.php

```diff
@@ -454,18 +475,58 @@ public static function update_widget( $widget_id, $sidebar, $position, $settings
 		if ( ! is_numeric( $position ) ) {
 			return new WP_Error( 'invalid_data', 'Invalid position', 400 );
 		}
+
+		// return if no widgets are present in the sidebar
 		$widgets_in_sidebar = self::get_widgets_in_sidebar( $sidebar );
 		if ( ! isset( $widgets_in_sidebar ) ) {
-			return new WP_Error( 'invalid_data', 'No such sidebar exists', 400 );
+			return new WP_Error( 'invalid_data', 'No widgets in sidebar', 400 );
+		}
+
+		// call when placing widget in a new sidebar, else obsolete
+		if ( ! self::is_widget_in_sidebar( $widgets_in_sidebar, $widget_id ) ) {
+			self::move_widget_to_sidebar( $widget, $sidebar, $position );
 		}
-		self::move_widget_to_sidebar( $widget, $sidebar, $position );
 		$widget_save_status = self::set_widget_settings( $widget_id, $settings );
 		if ( is_wp_error( $widget_save_status ) ) {
 			return $widget_save_status;
 		}
 		return self::get_widget_by_id( $widget_id );
 	}
 
+	/**
+	 * Retrieve the first active sidebar
+	 *
+	 * @return array|false Active sidebar or false if none exists
+	 */
+	public function get_first_sidebar() {
+		$active_sidebars = Jetpack_Widgets::get_active_sidebars();
+
+		unset( $active_sidebars[ 'array_version' ] );
+
+		if ( empty( $active_sidebars ) ) {
+			return false;
+		}
+		return array_shift( array_keys( $active_sidebars ) );
```

Comment:
This will throw a notice, because `array_shift` also updates the original referenced array (and there is no reference here).